src/Components/Web.JS/src/Boot.Server.ts

```diff
@@ -122,6 +122,63 @@ async function initializeConnection(options: CircuitStartOptions, logger: Logger
 
   Blazor._internal.forceCloseConnection = () => connection.stop();
 
+  Blazor._internal.sendJSDataStream = async (data: ArrayBufferView, streamId: string, maximumIncomingBytes: number) => {
+    // Run the rest in the background, without delaying the completion of the call to sendJSDataStream
+    // otherwise we'll deadlock (.NET can't begin reading until this completes, but it won't complete
+    // because nobody's reading the pipe)
+    setTimeout(async () => {
+      const maxMillisecondsBetweenAcks = 500;
+      let numChunksUntilNextAck = 5;
+      let lastAckTime = new Date().valueOf();
+      try {
+        // Chunk Size is the max SignalR message size less 512 bytes assumed overhead.
+        const chunkSize = maximumIncomingBytes - 512;
+        if (chunkSize <= 0) {
+          throw new Error('The chunk size must be greater than 0.')
+        }
+
+        let position = 0;
+
+        // Note: The server-side `StreamBufferCapacity` option (defaults to 10) can be configured to limit how many
+        // stream items from the client (per stream) will be stored before reading any more stream items (thus applying backpressure).
+        while (position < data.byteLength) {
+          const nextChunkSize = Math.min(chunkSize, data.byteLength - position);
+          console.log(`Sending a chunk of length ${nextChunkSize}`);
+          const nextChunkData = new Uint8Array(data.buffer, data.byteOffset + position, nextChunkSize);
+
+          numChunksUntilNextAck--;
+          if (numChunksUntilNextAck > 1) {
+            // Most of the time just send and buffer within the network layer
+            await connection.send('ReceiveJSDataChunk', streamId, nextChunkData, null);
```

Comment:
I wonder if we need a dedicated hub method or if this can be just piped through JS interop since we have an efficient way of sending bytes back and forth.

Reply:
One benefit of the dedicated hub method is that it can be private on the .NET side, whereas JS interop methods always have to be public.

I'm not fixated either way, but think this looks pretty much fine anyway.

Reply:
On the other hand, doing it through JS interop may mean we don't have to duplicate the implementation for WebAssembly and WebView too.

Reply:
That was my thinking.

Reply:
Okay consensus seems to be fine to leave as is, so to not require WASM/WebView specific implementations.